float32 phase1
float32 phase2
float32 phase3
float32 total